controllers/progressivesync_controller.go

```diff
@@ -35,6 +41,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/util/retry"
+	"k8s.io/kubernetes/pkg/util/hash"
```

Comment:
I'm not sure we should have a dependency on `k8s.io` because it's not supported. Why are we trying to do this?

Reply:
Hmm, what do you mean that it is not supported? If this is correct (https://pkg.go.dev/k8s.io/kubernetes/pkg/util/hash), there was a new version published recently.

The idea of using `DeepHashObject` was to pass the whole struct object and let it calculate the hash value from it.

Reply:
If you look at https://github.com/kubernetes/kubernetes/issues/81878#issuecomment-696689706 they say that using `k8s.io/kubernetes` is deprecated. Other projects are making an effort to remove that, for example https://github.com/argoproj/gitops-engine/issues/56 and https://github.com/argoproj/argo-cd/issues/5173

At the moment we have a depency on k8s because of argocd, once that is done we don't need that anymore.

Reply:
I see, ok, I will try and find a different way for us to calculate the hash value.

Reply:
If i read [this](https://github.com/argoproj-labs/applicationset/blob/master/api/v1alpha1/applicationset_types.go#L39) correctly, we will get a json from the `spec` [here](https://github.com/Skyscanner/applicationset-progressive-sync/pull/135/files#diff-a853c97824440aa3801077e1f5661f9087e27091372140f15f89ed5c8d0c5ba6R313). And then we can do something like this - https://play.golang.org/p/2clJUvGHJ_l. I will try it tomorrow and see if it works for us.

Reply:
This dependency is now removed and we rely on internal golang functions now.